Packages/Tracking/Core/Runtime/Plugins/LeapCSharp/Device.cs

```diff
@@ -255,65 +255,76 @@ public Pose DevicePose
+        internal Pose FindDeviceTransform()
+        {
+            // Check the service has valid support for device transforms
+            LEAP_VERSION minimumServiceVersion = new LEAP_VERSION { major = 5, minor = 19, patch = 0 };
+            if (!ServerStatus.IsServiceVersionValid(minimumServiceVersion))
+            {
+                devicePose = Pose.identity;
+                poseSet = true;
+                return Pose.identity;
+            }
 
-                deviceTransform = openXRToUnity * deviceTransform;
+            // Check the device transform is available before asking for one
+            bool deviceTransformAvailable = LeapC.GetDeviceTransformAvailable(Handle);
 
-                Vector3 outputPos = deviceTransform.GetPosition();
-                //Quaternion outputRot = deviceTransform.rotation; // Note: the matrices we receive are not rotatrion matrices. This produces unexpected results
+            if (!deviceTransformAvailable)
+            {
+                devicePose = Pose.identity;
+                return Pose.identity;
+            }
 
-                devicePose = new Pose(outputPos, Quaternion.identity);
+            // Get the device transform data and check it is valid as data
+            float[] data = new float[16];
+            eLeapRS result = LeapC.GetDeviceTransform(Handle, data);
 
-                poseSet = true;
-                return devicePose;
+            if (result != eLeapRS.eLeapRS_Success || data == null)
+            {
+                devicePose = Pose.identity;
+                return Pose.identity;
             }
+
+            // Using the LEAP->OPENXR device transform matrix
+            // Unitys matrices are generated as 4 columns:
+            Matrix4x4 deviceTransform = new Matrix4x4(
+                                                new Vector4(data[0], data[1], data[2], data[3]),
+                                                new Vector4(data[4], data[5], data[6], data[7]),
+                                                new Vector4(data[8], data[9], data[10], data[11]),
+                                                new Vector4(data[12], data[13], data[14], data[15]));
+
+            ////An example of the expected matrix if it were 8cm forward from the head origin
+            //// Unitys matrices are generated as 4 columns:
+            //Matrix4x4 deviceTransform = new Matrix4x4(
+            //                                    new Vector4(-0.001f, 0, 0, 0),
+            //                                    new Vector4(0, 0, -0.001f, 0),
+            //                                    new Vector4(0, -0.001f, 0, 0),
+            //                                    new Vector4(0, 0, -0.08f, 1));
+
+            //// An example of applying a rotation to the existing device transform
+            //Matrix4x4 rotationMatrix = Matrix4x4.Rotate(Quaternion.Euler(0, 90, 0));
+            //deviceTransform = deviceTransform * rotationMatrix;
```

Comment:
do we want to leave this commented out code in?

Reply:
I think it's nice to keep around for understanding _something_ about what's going on here?